trio/tests/test_util.py

```diff
@@ -58,3 +58,27 @@ def handler(signum, _):
                 async with ul1:
                     pass  # pragma: no cover
     assert "ul1" in str(excinfo.value)
+
+
+async def test_contextmanager_do_not_unchain_non_stopiteration_exceptions():
+    @acontextmanager
+    async def test_issue29692():
+        try:
+            yield
```

Comment:
Uh oh. Should I look more closely? `async_generator` is a pretty solid emulation of the built-in async generators, but it's had subtle bugs before....

One thing that just popped out at me looking again though: we surely want to replace all the `StopIteration`s in this patch with `StopAsyncIteration`s